### ISSUE TITLE
Improve package dependency conflict error message

### ISSUE THREAD

**Issue by negz** [crossplane/crossplane#4022]
### What problem are you facing?

I recently tried to install a provider with a package dependency on another provider. The package manager automatically tries to install the dependency if it does not exist. It names the new provider `{org}-{name}`. In this case `upbound-provider-gcp-config`.

I had accidentally already installed that same provider with a slightly different name - just `provider-gcp-config`. This caused a conflict - the package manager wasn't able to install the provider I wanted to install, or its dependency. This wasn't obvious to me at all due to the obscure error message it returned:

```
11s         Warning   ResolveDependencies      providerrevision/provider-gcp-datastore-dc66af0637b3             cannot resolve package dependencies: node already exists
```

### How could Crossplane help solve your problem?

Presuming this situation can only occur when a dependency conflicts with an existing package, it would be helpful for the error to state that.

**Comment by ytsarev** [comment]
Related https://github.com/crossplane/crossplane/issues/3423#issuecomment-1492086179

**Comment by ezgidemirel** [comment]
I've seen a similar error with the exact dependency name. I updated platform-ref-azure's `crossplane.yaml` file with the following changes and build this package -> xpkg.upbound.io/ezgiorg/ezgi-platform-ref-azure:v0.0.1

```
  dependsOn:
    - provider: xpkg.upbound.io/crossplane-contrib/provider-helm
      version: ">=v0.12.0"
    - provider: turkenh/provider-azure-dbforpostgresql
      version: 'v0.31.0-rc.1.12.gbc91102b'
    - provider: turkenh/provider-azure-config
      version: 'v0.31.0-rc.1.12.gbc91102b'
    - provider: turkenh/provider-azure-network
      version: 'v0.31.0-rc.1.12.gbc91102b'
    - provider: turkenh/provider-azure-containerservice
      version: 'v0.31.0-rc.1.12.gbc91102b'
```
As you can see `turkenh/provider-azure-config` is already in the list. It was installed successfully. However other providers that depend on provider-azure-config could not be installed. In providerrevisions I saw the following error message:

```
  Warning  ResolveDependencies  5s (x5 over 29s)   packages/providerrevision.pkg.crossplane.io  cannot resolve package dependencies: missing dependencies: [index.docker.io/turkenh/provider-azure-config]
```

**Comment by negz** [comment]
> I've seen a similar error with the exact dependency name.

Are you able to replicate this?

**Comment by turkenh** [comment]
> I've seen a similar error with the exact dependency name. I updated platform-ref-azure's crossplane.yaml file with the following changes and build this package -> xpkg.upbound.io/ezgiorg/ezgi-platform-ref-azure:v0.0.1

@ezgidemirel, I couldn't reproduce this with XP `v1.12.1`, please see below and let me know if I am missing something.

### Only install the configuration package

1. Install platform-ref-gcp with the following:

```yaml
apiVersion: pkg.crossplane.io/v1
kind: Configuration
metadata:
  name: platform-ref-gcp
spec:
  package: xpkg.upbound.io/upbound-release-candidates/platform-ref-gcp:v0.1.0
```

2. Wait for enough time until all packages are resolved and installed.

```bash
❯ kubectl get configurations.pkg.crossplane.io
NAME               INSTALLED   HEALTHY   PACKAGE                                                              AGE
platform-ref-gcp   True        True      xpkg.upbound.io/upbound-release-candidates/platform-ref-gcp:v0.1.0   4m39s

❯ kubectl get providers.pkg.crossplane.io
NAME                                                        INSTALLED   HEALTHY   PACKAGE                                                                             AGE
crossplane-contrib-provider-helm                            True        True      xpkg.upbound.io/crossplane-contrib/provider-helm:v0.14.0                            4m40s
upbound-release-candidates-provider-gcp-cloudplatform       True        True      xpkg.upbound.io/upbound-release-candidates/provider-gcp-cloudplatform:v0.32.0       4m36s
upbound-release-candidates-provider-gcp-compute             True        True      xpkg.upbound.io/upbound-release-candidates/provider-gcp-compute:v0.32.0             4m23s
upbound-release-candidates-provider-gcp-config              True        True      xpkg.upbound.io/upbound-release-candidates/provider-gcp-config:v0.32.0              3m39s
upbound-release-candidates-provider-gcp-container           True        True      xpkg.upbound.io/upbound-release-candidates/provider-gcp-container:v0.32.0           4m31s
upbound-release-candidates-provider-gcp-servicenetworking   True        True      xpkg.upbound.io/upbound-release-candidates/provider-gcp-servicenetworking:v0.32.0   4m1s
upbound-release-candidates-provider-gcp-sql                 True        True      xpkg.upbound.io/upbound-release-candidates/provider-gcp-sql:v0.32.0                 3m52s
```

### Install the base (i.e. config) package first

1. Install the base package with the following: 

```yaml
apiVersion: pkg.crossplane.io/v1
kind: Provider
metadata:
  name: provider-gcp-config
spec:
  package: xpkg.upbound.io/upbound-release-candidates/provider-gcp-config:v0.32.0
```

```
❯ kubectl get providers.pkg.crossplane.io
NAME                  INSTALLED   HEALTHY   PACKAGE                                                                  AGE
provider-gcp-config   True        True      xpkg.upbound.io/upbound-release-candidates/provider-gcp-config:v0.32.0   17s
```

2. Install the configuration package and wait for enough time:

```bash
❯ kubectl get configurations.pkg.crossplane.io
NAME               INSTALLED   HEALTHY   PACKAGE                                                              AGE
platform-ref-gcp   True        True      xpkg.upbound.io/upbound-release-candidates/platform-ref-gcp:v0.1.0   3m5s

❯ kubectl get providers.pkg.crossplane.io
NAME                                                        INSTALLED   HEALTHY   PACKAGE                                                                             AGE
crossplane-contrib-provider-helm                            True        True      xpkg.upbound.io/crossplane-contrib/provider-helm:v0.14.0                            3m8s
provider-gcp-config                                         True        True      xpkg.upbound.io/upbound-release-candidates/provider-gcp-config:v0.32.0              4m23s
upbound-release-candidates-provider-gcp-cloudplatform       True        True      xpkg.upbound.io/upbound-release-candidates/provider-gcp-cloudplatform:v0.32.0       3m4s
upbound-release-candidates-provider-gcp-compute             True        True      xpkg.upbound.io/upbound-release-candidates/provider-gcp-compute:v0.32.0             2m53s
upbound-release-candidates-provider-gcp-container           True        True      xpkg.upbound.io/upbound-release-candidates/provider-gcp-container:v0.32.0           3m
upbound-release-candidates-provider-gcp-servicenetworking   True        True      xpkg.upbound.io/upbound-release-candidates/provider-gcp-servicenetworking:v0.32.0   2m43s
upbound-release-candidates-provider-gcp-sql                 True        True      xpkg.upbound.io/upbound-release-candidates/provider-gcp-sql:v0.32.0                 2m33s
```

**Comment by ezgidemirel** [comment]
Hi @hasan, I tried to reproduce the issues with the latest release candidates and there were no installation problems. Here are the scenarios I tried:
- Added config package explicitly to the platform-ref-azure's dependsOn list. 
- Installed config package with the same name first and installed platform-ref-azure configuration
- Installed config package with a different name and installed platform-ref-azure configuration

Looks like the issue is solved.
